POEM_052.md

```diff
@@ -152,7 +241,32 @@ def some_func(x:{'units':'m', 'shape':4}=0.,
     return foo, bar
 ```
 Note: If `shape` metadata is given along with a non-scalar default value for the argument, then an error will be raised during setup by OpenMDAO. 
+Question: how do you indicate shape as an option, or function of options, like `num_nodes`? I would really hate to use a string, maybe we could do something with a context manager to provide a name, like:
 
+```python
+with om.func.options(num_nodes=int) as num_nodes:
```

Comment:
options for compoents can't be assumed to be known before setup time. So a context manager won't work. 

I think its much simpler to have options be additional named arguments to the function, or possibly allow **kwargs. We'll need some more annotation to cover any option meta-data. That detail will need to be worked out. We'll also need a specific means of identifying inputs vs options arguments to functions.